app/src/main/java/de/tum/in/tumcampusapp/api/tumonline/interceptors/CachingHelper.kt

```diff
@@ -0,0 +1,48 @@
+package de.tum.`in`.tumcampusapp.api.tumonline.interceptors
+
+import okhttp3.CacheControl
+import okhttp3.Response
+import org.joda.time.Duration
+import java.util.concurrent.TimeUnit
+
+class CachingHelper {
+
+    fun isCacheable(url: String) = getCachingDuration(url) != Duration.ZERO
+
+    fun getCachingDuration(url: String): Duration {
+        return cachingDurations
+                .entries
+                .filter { url.contains(it.key) }
+                .map { it.value }
+                .getOrElse(0) { Duration.ZERO }
+    }
+
+    fun updateCacheControlHeader(url: String, response: Response): Response {
+        val duration = getCachingDuration(url)
+
+        val maxAge = duration.toStandardDays().days
+        val cacheControl = CacheControl.Builder()
+                .maxAge(maxAge, TimeUnit.DAYS)
+                .build()
+
+        return response.newBuilder()
+                .removeHeader("Cache-Control")
+                .addHeader("Cache-Control", cacheControl.toString())
+                .build()
+    }
+
+    companion object {
+
+        private val cachingDurations = mapOf(
```

Comment:
When we're not actually using the Map capabilities of that map, because we always linearly search the map, please use a List of tuples instead of a Map.

Reply:
Done.